src/webots/maths/WbAxisAngle.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright 1996-2020 Cyberbotics Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef WB_AXIS_ANGLE_HPP
+#define WB_AXIS_ANGLE_HPP
+
+#include "WbVector3.hpp"
+
+//
+// Description: Axis-angle representation
+//
+
+class WbAxisAngle {
+public:
+  WbAxisAngle(double x, double y, double z, double angle) : mAxis(WbVector3(x, y, z)), mAngle(angle){};
+
+  WbVector3 &axis() { return mAxis; };
```

Comment:
I don't think can add `const` here because `mAxis` is returned as a reference. That means we can alternate `mAxis`, effectively changing `WbAxisAngle`?

Reply:
OK, I was not sure about it. So, it's fine this way. Other wise we may return a const WbVector3:
`const WbVector3 &axis() const { return mAxis; };

Reply:
I am modifying the `axis()` output here:
https://github.com/cyberbotics/webots/pull/2424/files#diff-f711b25b459281533b960b951ada0d487fdcd2200884b1e6f495da1209fe7ccbR149

The only solution would be to wrap the methods, `setX`, `setY`, `setZ` and `normalize`

Reply:
OK, then keep it as it is. It's fine.